.github/workflows/dependabot.js.yml

```diff
@@ -0,0 +1,41 @@
+name: Dependabot
+
+on:
+  push:
+    branches: [ dependabot/npm_and_yarn/** ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Use Node.js 12.x
+      uses: actions/setup-node@v1
+      with:
+        node-version: '12'
+    - name: Autofix lockfile
+      run: |
+        # restore yarn.lock from the previous commit
+        git checkout HEAD^ -- yarn.lock
+
+        # install yarn-plugin-deduplicate
+        yarn plugin import https://raw.githubusercontent.com/eps1lon/yarn-plugin-deduplicate/latest/bin/%40yarnpkg/plugin-deduplicate.js
+
+        # if package.json was not updated, upgrade the dependency
+        git diff --name-only HEAD^ HEAD | grep -q 'package.json' || yarn up `git log -1 --pretty=%s | awk '{ print $3 }'`
+
+        # restore package.json from the last commit
+        git checkout HEAD -- package.json
+
+        yarn install
+
+        # deduplicate lockfile
+        yarn deduplicate
```

Comment:
I'm kind of surprised this is default behaviour built into `yarn`.  Do the other commands in this step making duplication more of an issue than usual?

Reply:
@kevinbarabash Yarn doesn't dedupe dependencies when upgrading. For instance, if we have
```json
{
  "dependencies": {
    "A": "^1.2.0"
  }
}
```
```yaml
A@^1.1.0, A@^1.2.0
  version: "1.2.1"
```
and run `yarn up A`, it gives
```yaml
A@^1.1.0
  version: "1.2.1"

A@^1.2.0
  version: "1.3.0"
```

Reply:
TIL 🙂